### PR TITLE
Replace `throw new ArgumentOutOfRangeException` with `ArgumentOutOfRangeException.ThrowIf...`

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
Replaces 
```cs
            if (start < 0)
            {
                throw new ArgumentOutOfRangeException("start");
            }
```
with
```cs
ArgumentOutOfRangeException.ThrowIfNegative(start);
```

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.Management.UI.Internal/HelpWindow/ParagraphBuilder.cs [hunk 1]
```diff
@@ -276,15 +276,8 @@ private static void MoveSpanToPosition(ref int currentSpanIndex, ref TextSpan? c
...
-            if (start < 0)
-            {
-                throw new ArgumentOutOfRangeException("start");
-            }
-
-            if (start + length > this.textBuilder.Length)
-            {
-                throw new ArgumentOutOfRangeException("length");
-            }
+            ArgumentOutOfRangeException.ThrowIfNegative(start);
```
Comment (PaulHigin) [inline src/Microsoft.Management.UI.Internal/HelpWindow/ParagraphBuilder.cs:279]: I don't understand why the parameter name is no longer provided:
```suggestion
            ArgumentOutOfRangeException.ThrowIfNegative(start, nameof(start));
```

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: src/Microsoft.Management.UI.Internal/HelpWindow/ParagraphBuilder.cs (hunk 1 above)
Comment (CarloToso) [reply src/Microsoft.Management.UI.Internal/HelpWindow/ParagraphBuilder.cs:279]: It's no longer needed:
```cs
int start = -1;

if (start < 0)
{
	throw new ArgumentOutOfRangeException("start");
}
// Unhandled exception. System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter 'start')

ArgumentOutOfRangeException.ThrowIfNegative(start);
// Unhandled exception. System.ArgumentOutOfRangeException: 'start' must be a non-negative value. (Parameter 'start')
// Actual value was -1.
```

**Review by PaulHigin — APPROVED** [review]

**Comment by PaulHigin** [comment]
@CarloToso Thanks!

**Comment by xtqqczze** [comment]
@PaulHigin The merge commit doesn't have a useful description. I believe it is best practice to use `Squash and merge` to use the description from the pull request.

**Pull request merged**